src/libs/residfp/resample/SincResampler.cpp

```diff
@@ -332,15 +332,17 @@ SincResampler::SincResampler(double clockFrequency, double samplingFrequency, do
         // Calculate the sinc tables.
         const double scale = 32768.0 * wc / cyclesPerSampleD / M_PI;
 
+        const double firN_2 = static_cast<double>(firN / 2);
```

Comment:
Is this a micro-pre-optimization, or more of a "make a variable anytime there's a repeated expression"?

Reply:
Reported this upstream https://github.com/libsidplayfp/libsidplayfp/issues/28

In the original math expression, the `firN / 2` doesn't retain it's fractional part, but the result is still treated as a float-point value, which can indicate an oversight. So The residfp author made this explicit/clear :-)